plugins/uefi/fu-uefi-pcrs.c

```diff
@@ -103,10 +96,52 @@ fu_uefi_pcrs_setup_tpm12 (FuUefiPcrs *self, const gchar *fn_pcrs, GError **error
 }
 
 static gboolean
-fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, const gchar *argv0, GError **error)
+fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, GError **error)
 {
-	const gchar *argv[] = { argv0, NULL };
-	return fu_common_spawn_sync (argv, fu_uefi_pcrs_parse_line, self, 1500, NULL, error);
+	TSS2_RC rc;
+	ESYS_CONTEXT *ctx = NULL;
+	TPML_PCR_SELECTION pcr_selection_in = { 0, };
+	TPML_DIGEST *pcr_values = NULL;
+	g_autoptr(GString) str = NULL;
+	FuUefiPcrItem *item;
+
+	rc = Esys_Initialize (&ctx, NULL, NULL);
+	if (rc != TSS2_RC_SUCCESS) {
+		return FALSE;
+	}
+
+	/* currently we only use PCR 0 on banks SHA-1 and SHA-256, so fetch only these */
```

Comment:
Who is we in this context?  Do you mean generally only SHA1 and SHA256 are used for PCR0?
I would wonder if we can future proof this a little further for SHA384 and SHA512.  Does tpm2-tss support those yet?